src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,106 @@
+import { HttpClient, HttpEvent, HttpEventType } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { apiSettings } from '../../environments/apisettings';
+import { MoreResults, RecordPublicationInfo, Results, SortParameter } from '../_models';
+
+import { collectResultsUptoPage } from './service-utils';
+import { ErrorService } from './error.service';
+
+@Injectable({ providedIn: 'root' })
+export class DepublicationService {
+  constructor(private readonly http: HttpClient, private readonly errors: ErrorService) {}
```

Comment:
Oh, and one thing... I think the httpClient takes care of authentication, right? The requests in this class will need the same authentication token in the header as the other requests.

Reply:
I think it will work as is - based on what I see working for the dataset form